frontend/src/app/aethel/aethel.component.ts

```diff
@@ -127,19 +131,35 @@ export class AethelComponent implements OnInit {
     }
 
     private updateUrl(queryInput: AethelInput): void {
-        // This does not actually refresh the page because it just adds parameters to the current route.
-        // This triggers a new query.
+        // This does not actually refresh the page because it just adds
+        // parameters to the current route. This triggers a new query.
         const url = this.router
             .createUrlTree([], {
                 relativeTo: this.route,
-                queryParams: {
-                    ...queryInput,
-                },
+                queryParams: this.formatQueryParams(queryInput),
             })
             .toString();
         this.router.navigateByUrl(url);
     }
 
+    private formatQueryParams(queryInput: AethelInput): AethelInput {
+        const queryParams: AethelInput = {};
+
+        // Only include word and type in the URL if they are not null,
+        // undefined or an empty string.
+        if (queryInput.word && queryInput.word !== "") {
```

Comment:
This is to avoid ugly URLs like `http://localhost:4200/aethel?word=&type=VNW&limit=10&skip=0`, where there is no value for `word`. I assume `underscore` or `lodash` has a nice util for this, but this works.